src/components/inline_edit/inline_edit_form.test.tsx

```diff
@@ -0,0 +1,303 @@
+            color: 'primary',
+            iconSide: 'left',
+          }}
+        />
+      );
+
+      expect(container.firstChild).toMatchSnapshot();
+      expect(getByTestSubject('euiInlineReadModeButton')).toBeTruthy();
+    });
+
+    it('renders small size', () => {
+      const { container } = render(
+        <EuiInlineEditForm
+          {...commonInlineEditFormProps}
+          sizes={SMALL_SIZE_FORM}
+        />
+      );
+
+      expect(container.firstChild).toMatchSnapshot();
+    });
+  });
+
+  describe('Edit Mode', () => {
+    it('renders', () => {
+      const { container } = render(
+        <EuiInlineEditForm
+          {...commonInlineEditFormProps}
+          startWithEditOpen={true}
+        />
+      );
+
+      expect(container.firstChild).toMatchSnapshot();
+    });
+
+    it('renders editModeProps.inputProps', () => {
+      const { container, getByTestSubject } = render(
+        <EuiInlineEditForm
+          {...commonInlineEditFormProps}
+          startWithEditOpen={true}
+          editModeProps={{
+            inputProps: {
+              prepend: 'Prepend Example',
+              'data-test-subj': 'customInput',
+            },
+          }}
+        />
+      );
+
+      expect(container.firstChild).toMatchSnapshot();
+      expect(getByTestSubject('customInput')).toBeTruthy();
+    });
+
+    it('renders editModeProps.formRowProps', () => {
+      const { container, getByTestSubject } = render(
+        <EuiInlineEditForm
+          {...commonInlineEditFormProps}
+          startWithEditOpen={true}
+          editModeProps={{
+            formRowProps: {
+              error: ['This is an error'],
+              'data-test-subj': 'customErrorText',
+            },
+          }}
+        />
+      );
+
+      expect(container.firstChild).toMatchSnapshot();
+      expect(getByTestSubject('customErrorText')).toBeTruthy();
+    });
+
+    it('renders save button and cancel button aria-labels', () => {
+      const { container, getByLabelText } = render(
+        <EuiInlineEditForm
+          {...commonInlineEditFormProps}
+          startWithEditOpen={true}
+          saveButtonAriaLabel="Yes! Let's save."
+          cancelButtonAriaLabel="Uh no. Do not save."
+        />
+      );
+
+      expect(container.firstChild).toMatchSnapshot();
+      expect(getByLabelText("Yes! Let's save.")).toBeTruthy();
+      expect(getByLabelText('Uh no. Do not save.')).toBeTruthy();
+    });
+
+    it('renders EuiSkeletonRectangles in place of editMode buttons when loading', () => {
+      const { container, getByTestSubject, queryByTestSubject } = render(
+        <EuiInlineEditForm
+          {...commonInlineEditFormProps}
+          startWithEditOpen={true}
+          isLoading={true}
+        />
+      );
+
+      expect(container.firstChild).toMatchSnapshot();
+
+      expect(getByTestSubject('euiInlineEditModeSaveLoading')).toBeTruthy();
+      expect(getByTestSubject('euiInlineEditModeCancelLoading')).toBeTruthy();
```

Comment:
see above comment re: not needing a `data-test-subj` attr for this
```suggestion
      expect(container.querySelectorAll('.euiSkeletonRectangle')).toHaveLength(2);
```